### ISSUE TITLE
Compile zabbix_docker_module for Zabbix 4.0 LTS

### ISSUE THREAD

**Issue by prairiewolf-by** [monitoringartist/zabbix-docker-monitoring#118]
Hello. After couple of unsuccessful compilations, I wanna try to open an issue, maybe you help the community.
What is a goal: monitor docker containers.
Zabbix agent version: 4.0.5
Zabbix server version: 4.0.5
I followed instructions and faced same as here: https://github.com/monitoringartist/zabbix-docker-monitoring/issues/116. (Issue was closed without resolution or any help, what seems to be inappropriate).
But that module i've compiled is **not** working - Zabbix agent does not start with included module compiled.
Those compilation warnings seems to be crucial for module to work as designed.
Or there is no support in module code for Zabbix v4 LTS - just announce this, if this is true.

Thanks.

**Comment by jangaraj** [comment]
> I followed instructions and faced same as here: #116. (Issue was closed without resolution or any help, what seems to be inappropriate).

There were only compilation warnings, no errors. Check gcc configuration

> But that module i've compiled is not working - Zabbix agent does not start with included module compiled.

Any logs and used OS, gcc version, compilation commands, please? I can't replicate your problem without any input.

**Comment by prairiewolf-by** [comment]
@jangaraj , thanks a lot for feedback.

My OS is Ubuntu 18.04.2 LTS
gcc version 7.3.0 (Ubuntu 7.3.0-27ubuntu1~18.04)

Commands executed and their logs attached.
[command_logs.txt](https://github.com/monitoringartist/zabbix-docker-monitoring/files/3017417/command_logs.txt)

After that i copied module to /usr/lib/zabbix/modules/ and chmod'ed and chown'ed it to give necessary permissions.
That gave no effect in any combinations.

Any more input from my side?

**Comment by jangaraj** [comment]
I don't see any error in the compilation - only warnings. I guess you need https://github.com/monitoringartist/zabbix-docker-monitoring#troubleshooting

**Comment by prairiewolf-by** [comment]
Hey @jangaraj , I am very thankful for feedback and help.
As far as I have just compiled it for the first time, my questions may have seen as dummy. :)
The problem is resolved, I have placed module into /usr/local/lib instead of usr/lib and fixed mistypo in filename.
The module was up, but server received no data / no discovery - and then i restarted zabbix server.
It seems data began going to server.
![image](https://user-images.githubusercontent.com/26344498/55154491-03932600-5166-11e9-8fcb-a5296f1ad999.png)

In overall, compilation guide is 100% clear and working - it is just necessary to keep an eye over filenames and permissions, along with possible server behavior issues.
